app/front/scripts/components/babbage-geoview/utils.js

```diff
@@ -76,35 +80,74 @@ function updateDimensions(geoJson, options) {
   geoJson.unscaledBounds = path.bounds(geoJson);
 }
 
-function updateValues(geoJson, options) {
-  var sum = 0;
+function findMedianRange(values) {
+  var radius = Math.ceil(values.length / 5);
   var min = null;
   var max = null;
+
+  if (values.length > 0) {
```

Comment:
This algorithm is used to find min and max value for coloring range. Before this fix I just used min and max from data array. But it's very common situation when array contains several values that are 100 or more times larger then other values. In this case, we'll have few fully colored regions on map, and other ones will be desaturated (because all of them has 100 times less values, comparing to max). So this function applies [median filter](https://en.wikipedia.org/wiki/Median_filter) with adaptive window to array of values, and then takes min and max values. This makes min/max range much closer to most values in array. I'll add comment in sources for that.